src/grbindex/grbindex.f

```diff
@@ -1,81 +1,81 @@
-C>  - GETARG       GET COMMAND LINE ARGUMENT
-C>  - WRGI1H       WRITE INDEX HEADERS
-C>  - GETGIR       GET INDEX BUFFER
-C>  - BAWRITE      BYTE-ADDRESSABLE WRITE
-C>  - ERRMSG       WRITE A MESSAGE TO STDERR
-C>  - ERREXIT      EXIT WITH RETURN CODE
-C>
-C> EXIT STATES:
-C>  - COND =   0 - SUCCESSFUL RUN
-C>  - COND =   1 - GRIB MESSAGE NOT FOUND
-C>  - COND =   2 - INCORRECT ARGUMENTS
-C>  - COND =   8 - ERROR ACCESSING FILE
-C>
+!> @file
+!> Write an index file.
+!> @author IREDELL @date 1998-09-15
+!
+!> PROGRAM CREATES AN INDEX FILE FROM A GRIB FILE.
+!>   THE INDEX FILE SERVES AS A TABLE OF CONTENTS FOR THE GRIB FILE,
+!>   ENABLING QUICK ACCESS TO THE DATA.  THE GRIB FILE MUST BE UNBLOCKED,
+!>   BUT THERE CAN BE A GAP BEFORE THE FIRST GRIB MESSAGE OF AT MOST
+!>   32000 BYTES AND GAPS BETWEEN MESSAGES OF AT MOST 4000 BYTES.
+!>   THE TWO FILE NAMES ARE RETRIEVED FROM THE COMMAND LINE ARGUMENTS.
+!>   THE FIRST ARGUMENT IS THE NAME OF THE INPUT GRIB FILE.
+!>   THE SECOND ARGUMENT IS THE NAME OF THE OUTPUT INDEX FILE.
+!>   CURRENTLY, ONLY VERSION 1 OF GRIB CAN BE READ.
+!>   VERSION 1 OF THE INDEX FILE HAS THE FOLLOWING FORMAT:
+!>     81-BYTE S.LORD HEADER WITH 'GB1IX1' IN COLUMNS 42-47 FOLLOWED BY
+!>     81-BYTE HEADER WITH NUMBER OF BYTES TO SKIP BEFORE INDEX RECORDS,
+!>     NUMBER OF BYTES IN EACH INDEX RECORD, NUMBER OF INDEX RECORDS,
+!>     AND GRIB FILE BASENAME WRITTEN IN FORMAT ('IX1FORM:',3I10,2X,A40).
+!>     EACH FOLLOWING INDEX RECORD CORRESPONDS TO A GRIB MESSAGE
+!>     AND HAS THE INTERNAL FORMAT:
+!>     -  BYTE 001-004: BYTES TO SKIP IN DATA FILE BEFORE GRIB MESSAGE
+!>     -  BYTE 005-008: BYTES TO SKIP IN MESSAGE BEFORE PDS
+!>     -  BYTE 009-012: BYTES TO SKIP IN MESSAGE BEFORE GDS (0 IF NO GDS)
+!>     -  BYTE 013-016: BYTES TO SKIP IN MESSAGE BEFORE BMS (0 IF NO BMS)
+!>     -  BYTE 017-020: BYTES TO SKIP IN MESSAGE BEFORE BDS
+!>     -  BYTE 021-024: BYTES TOTAL IN THE MESSAGE
+!>     -  BYTE 025-025: GRIB VERSION NUMBER
+!>     -  BYTE 026-053: PRODUCT DEFINITION SECTION (PDS)
+!>     -  BYTE 054-095: GRID DEFINITION SECTION (GDS) (OR NULLS)
+!>     -  BYTE 096-101: FIRST PART OF THE BIT MAP SECTION (BMS) (OR NULLS)
+!>     -  BYTE 102-112: FIRST PART OF THE BINARY DATA SECTION (BDS)
+!>     -  BYTE 113-172: (OPTIONAL) BYTES 41-100 OF THE PDS
+!>     -  BYTE 173-184: (OPTIONAL) BYTES 29-40 OF THE PDS
+!>     -  BYTE 185-320: (OPTIONAL) BYTES 43-178 OF THE GDS
+!>
+!> PROGRAM HISTORY LOG:
+!> -  92-11-22  IREDELL
+!> -  94-06-08  EBISUZAKI - ELIMINATE ISHELL CALLS
+!> -  94-08-26  IREDELL   - 40 BYTE PDS EXTENSION
+!> -  95-10-31  IREDELL   - CONSIDERABLY REDUCE I/O
+!> -  96-10-31  IREDELL   - AUGMENTED OPTIONAL DEFINITIONS TO BYTE 320
+!> - 1999-04-27  GILBERT   - Changed CALL EXIT(N) to CALL ERREXIT(N) so that
+!>                         the proper error return value is passed back to
+!>                         the shell.
+!> - 2012-07-31  VUONG     - CHANGED HOSTNAME TO HOSTNAM
```

Comment:
This isn't consistent with our other PHL's.

Reply:
Fixed.